packages/test/test-service-load/src/loadTestDataStore.ts

```diff
@@ -574,10 +574,23 @@ class LoadTestDataStore extends DataObject implements ILoadTest {
 				? Math.floor(Math.random() * opSizeinBytes)
 				: opSizeinBytes;
 		const largeOpRate = config.testConfig.content?.largeOpRate ?? 1;
+		// To avoid having all clients send their large payloads at roughly the same time
+		const largeOpJitter = Math.floor(Math.random() * largeOpRate);
+		// To avoid growing the file size unnecessarily, not all clients should be sending large ops
+		const maxClientsSendingLargeOps = config.testConfig.content?.numClients ?? 1;
 		let opsSent = 0;
 
 		const sendSingleOp = () => {
-			if (opSizeinBytes > 0 && largeOpRate > 0 && opsSent % largeOpRate === 1) {
+			if (
+				this.shouldSendLargeOp(
+					opSizeinBytes,
+					largeOpRate,
+					opsSent,
+					largeOpJitter,
+					config.runId,
+					maxClientsSendingLargeOps,
+				)
+			) {
 				dataModel.sharedmap.set(`key${opsSent}`, generateContentOfSize(getOpSizeInBytes()));
```

Comment:
is this key still the problem. i don't know why we would ever need more than a single key. at most i think a key per runner or runId

Reply:
Sure, but I don't believe this to be a problem (at least not right now). There will be 10 map keys created for a run. Is that problematic?

Reply:
i don't know. its hard for me to understand the interplay of all the settings, and if there is a bad config matrix, as ideally any config works, it just does more or less. we are not there today, but the fewer traps lying around the better

Reply:
to me this feature isn't trying to grow the snapshot, so it should avoid doing it more than it needs to